### PR TITLE
make download() more predictable, fewer side effects

### PR DESCRIPTION
**Pull request by andreasjansson**

### CODE REVIEW

File: mirdata/beatles.py [hunk 1]
```diff
@@ -22,46 +22,22 @@
...
+            > {beatles_dir}/
+                > annotations/
+                > audio/
+        and copy the {beatles_dir} folder to {save_path}
+    """.format(
+            beatles_dir=BEATLES_DIR, save_path=save_path
         )
-        return
-
-    download_path = utils.download_from_remote(
```
Comment (rabitt) [inline mirdata/beatles.py]: this should stay in - it's downloading the annotations from the beatles' dataset website

File: mirdata/orchset.py [hunk 2]
```diff
@@ -50,26 +50,22 @@ def download(data_home=None, force_overwrite=False):
     save_path = utils.get_save_path(data_home)
     dataset_path = os.path.join(save_path, ORCHSET_DIR)
 
+    if exists(data_home) and not force_overwrite:
```
Comment (rabitt) [inline mirdata/orchset.py]: `data_home` is the directory above the dataset

File: mirdata/orchset.py [hunk 3]
```diff
@@ -50,26 +50,22 @@ def download(data_home=None, force_overwrite=False):
...
+        return
+
     if force_overwrite:
         utils.force_overwrite_all(ORCHSET_META, dataset_path, data_home)
```
Comment (rabitt) [inline mirdata/orchset.py]: we should rename `force_overwrite_all` to `force_delete_all`

File: mirdata/orchset.py [hunk 4]
```diff
@@ -50,26 +50,22 @@ def download(data_home=None, force_overwrite=False):
...
-        return
-
     download_path = utils.download_from_remote(
         ORCHSET_META, force_overwrite=force_overwrite
     )
     utils.unzip(download_path, save_path, dataset_path)
-    validate(dataset_path, data_home)
+
+
+def exists(data_home=None):
```
Comment (rabitt) [inline mirdata/orchset.py]: let's abstract this into utils.

File: mirdata/salami.py [hunk 5]
```diff
@@ -39,46 +39,21 @@
...
+            > {salami_dir}/
+                > salami-data-public-master/
+                > audio/
+        and copy the {salami_dir} folder to {save_path}
+    """.format(
+            salami_dir=SALAMI_DIR, save_path=save_path
         )
-        return
-
-    download_path = utils.download_from_remote(
```
Comment (rabitt) [inline mirdata/salami.py]: this should stay

File: mirdata/orchset.py (hunk 2 above)
Comment (andreasjansson) [reply mirdata/orchset.py]: this is actually intentional at the moment (`exists` builds the actual path). will change in future pr.

**Pull request merged**